### PR TITLE
Add editor toolbar to emailbody element - backport 3.9

### PR DESCRIPTION
**Pull request by glepod**

### CODE REVIEW

File: version.php [hunk 1]
```diff
@@ -24,8 +24,8 @@
 
 defined('MOODLE_INTERNAL') || die;
 
-$plugin->version   = 2021100103;
-$plugin->release   = '2021100103';
+$plugin->version   = 2023032100;
```
Comment (danmarsden) [inline version.php]: Hi @glepod - looks like you need to take a look at the version bump rules :-) - in the latest development branch you bump the full version date, but in older stable branches you only bump by one. more info here:
https://moodledev.io/docs/apis/commonfiles/version.php#version

the rest seems ok to me though.

Comment (glepod) [reply version.php]: Hi @danmarsden - Thanks for the pickup - I missed that :)

File: classes/task/check_recompletion.php [hunk 2]
```diff
@@ -149,15 +149,22 @@ protected function notify_user($userid, $course, $config) {
...
-            } else {
-                // This is most probably the tag/newline soup known as FORMAT_MOODLE.
-                $messagehtml = format_text($message, FORMAT_MOODLE, array('context' => $context,
+            if (property_exists($config, "recompletionemailbody_format")) {
+                $messagehtml = format_text($message, FORMAT_HTML, array('context' => $context,
                     'para' => false, 'newlines' => true, 'filter' => true));
                 $messagetext = html_to_text($messagehtml);
+            } else {
+                // Handle older data record created with textarea element.
+                if (strpos($message, '<') === false) {
```
Comment (brendanheywood) [inline classes/task/check_recompletion.php]: Can you store the format properly with an upgrade script so we don't have this logic?

**Comment by glepod** [comment]
Hi Dan. I Have updated this pr with the latest changes from 311_STABLE.

**Comment by danmarsden** [comment]
Thanks Glen - all merged in! :-)

**Pull request merged**